python/python/tests/test_indices.py

```diff
@@ -10,16 +10,25 @@
 NUM_ROWS = 10000
 DIMENSION = 128
 NUM_SUBVECTORS = 8
-NUM_PARTITIONS = 100
+NUM_FRAGMENTS = 3
+NUM_PARTITIONS = round(np.sqrt(NUM_FRAGMENTS * NUM_ROWS))
 
 
 @pytest.fixture(params=[np.float16, np.float32, np.float64], ids=["f16", "f32", "f64"])
 def rand_dataset(tmpdir, request):
     vectors = np.random.randn(NUM_ROWS, DIMENSION).astype(request.param)
     vectors.shape = -1
     vectors = pa.FixedSizeListArray.from_arrays(vectors, DIMENSION)
-    table = pa.Table.from_arrays([vectors], names=["vectors"])
-    ds = lance.write_dataset(table, str(tmpdir / "dataset"))
+    uri = str(tmpdir / "dataset")
+
+    fragments = []
+    for i in range(NUM_FRAGMENTS):
+        table_append = pa.Table.from_arrays([vectors], names=["vectors"])
+        fragment = lance.fragment.LanceFragment.create(uri, table_append)
+        fragments.append(fragment)
+
+    operation = lance.LanceOperation.Overwrite(table_append.schema, fragments)
+    ds = lance.LanceDataset.commit(uri, operation)
```

Comment:
A simpler way to create a dataset with `NUM_FRAGMENTS` is to change the constants:

```
NUM_ROWS_PER_FRAGMENT = 10000
NUM_FRAGMENTS = 3
NUM_ROWS = NUM_FRAGMENTS * NUM_ROWS_PER_FRAGMENT
```

Then this becomes:

```
ds = lance.write_dataset(table, str(tmpdir / "dataset"), max_rows_per_file=NUM_ROWS_PER_FRAGMENT)
```